### PR TITLE
OAuth Working

### PR DESCRIPTION
**Pull request by andreayang310**
Not sure how this is gonna interface with the UI yet but here's what I have so far (it's also deployed because that was my only way of testing it).
I had to change the user page url for it to work with the web.xml mapping, so now when you click "your page" is when the authorization servlet gets called.

### CODE REVIEW

File: src/main/java/com/google/codeu/servlets/AboutMeServlet.java [hunk 1]
```diff
@@ -67,6 +67,6 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     User user = new User(userEmail, aboutMe);
     datastore.storeUser(user);
 
-    response.sendRedirect("/user-page.html?user=" + userEmail);
+    response.sendRedirect("u/user-page.html?user=" + userEmail);
```
Comment (ctrokid) [inline src/main/java/com/google/codeu/servlets/AboutMeServlet.java:69]: Is this necessary?

File: src/main/java/com/google/codeu/servlets/CalendarOAuthServlet.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CalendarOAuthServlet extends AbstractAppEngineAuthorizationCodeServlet {
+
+  @Override
+  protected void doGet(HttpServletRequest request, HttpServletResponse response)
+          throws IOException {
+    GoogleAuthorizationCodeFlow flow = Utils.newFlow();
```
Comment (ctrokid) [inline src/main/java/com/google/codeu/servlets/CalendarOAuthServlet.java:26]: A high level comment here would be useful for teammates to understand what's going on. Nothing too detailed, but a little more explicit than what the code says.

File: src/main/java/com/google/codeu/servlets/LoginServlet.java [hunk 3]
```diff
@@ -28,7 +28,7 @@
 /**
  * Redirects the user to the Google login page or their page if they're already logged in.
  */
-@WebServlet("/login")
+//@WebServlet("/login")
```
Comment (ctrokid) [inline src/main/java/com/google/codeu/servlets/LoginServlet.java]: Remove :)

File: src/main/java/com/google/codeu/servlets/LoginStatusServlet.java [hunk 4]
```diff
@@ -29,7 +29,7 @@
 /**
  * Returns login data as JSON, e.g. {"isLoggedIn": true, "username": "Ada"}
  */
-@WebServlet("/login-status")
+//@WebServlet("/login-status")
```
Comment (ctrokid) [inline src/main/java/com/google/codeu/servlets/LoginStatusServlet.java]: Remove

File: src/main/java/com/google/codeu/servlets/LogoutServlet.java [hunk 5]
```diff
@@ -27,13 +27,13 @@
 /**
  * Redirects the user to the Google logout page, which then redirects to the homepage.
  */
-@WebServlet("/logout")
+//@WebServlet("/logout")
```
Comment (ctrokid) [inline src/main/java/com/google/codeu/servlets/LogoutServlet.java]: Remove

File: src/main/java/com/google/codeu/servlets/OAuth2Callback.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    String user = userService.getCurrentUser().getEmail();
+    resp.sendRedirect("/user-page.html?user=" + user);
+  }
+
+  @Override
+  protected void onError(HttpServletRequest req, HttpServletResponse resp,
+                         AuthorizationCodeResponseUrl errorResponse)
+          throws ServletException, IOException {
+    String nickname = UserServiceFactory.getUserService().getCurrentUser().getNickname();
+    resp.getWriter().print("<h3>" + nickname + ", why don't you want to play with me?</h1>");
```
Comment (ctrokid) [inline src/main/java/com/google/codeu/servlets/OAuth2Callback.java:33]: Lol. Feel free to update this, or maybe discuss with the team what you want to do in this situation.

File: src/main/java/com/google/codeu/servlets/Utils.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+  /** Global instance of the JSON factory. */
+  static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+  private static GoogleClientSecrets clientSecrets = null;
+
+  static GoogleClientSecrets getClientCredential() throws IOException {
+    if (clientSecrets == null) {
+      clientSecrets = GoogleClientSecrets.load(JSON_FACTORY,
+              new InputStreamReader(Utils.class.getResourceAsStream("/client_secrets.json")));
```
Comment (ctrokid) [inline src/main/java/com/google/codeu/servlets/Utils.java:45]: Really nice! How did you deploy this file to appengine?

Oh.. I think I see that it's committed below to Github. This isn't ideal but I don't blame you for doing this. Let me check into this real quick to see if we can avoid this.

File: src/main/java/com/google/codeu/servlets/AboutMeServlet.java (hunk 1 above)
Comment (ctrokid) [reply src/main/java/com/google/codeu/servlets/AboutMeServlet.java:69]: Oh I see. Nice! I like this.

File: src/main/java/com/google/codeu/servlets/MessageFeedServlet.java [hunk 8]
```diff
@@ -33,6 +33,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     response.setContentType("application/json");
 
     List<Message> messages = datastore.getAllMessages();
+    //try calling calendar here
```
Comment (ctrokid) [inline src/main/java/com/google/codeu/servlets/MessageFeedServlet.java:36]: Not sure why Github won't let me comment on line 16 above. But is the `@WebServlet("/feed")` working? I don't believe I see the mapping in `web.xml`.

If it is working, that's pretty interesting. Combining both servlet mappings in `web.xml` and via annotations didn't seem possible when I was trying this 😄. I would suggest being consistent and moving that mapping into `web.xml` regardless of whether this is working right now.

**Review by ctrokid — COMMENTED** [review]
Great job! This looks really good and I'm looking forward to logging in on the app.

File: src/main/java/com/google/codeu/servlets/Utils.java (hunk 7 above)
Comment (andreayang310) [reply src/main/java/com/google/codeu/servlets/Utils.java:45]: I believe we can keep this out of the commit by setting environment variables instead? I can look into it when I have sec.

Comment (ctrokid) [reply src/main/java/com/google/codeu/servlets/Utils.java:45]: That SGTM, but also due to time constraints, I'm also okay with this just getting submitted to GitHub. 😦 It is up to you.

**Review by ctrokid — APPROVED** [review]
Approved!

**Pull request merged**